ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/Enclose.hs

```diff
@@ -32,3 +35,18 @@ encloseWith ::
   -> m a
 encloseWith tracer action =
     traceWith tracer RisingEdge *> action <* traceWith tracer FallingEdge
+
+type EnclosingTimed = Enclosing' DiffTime
+
+encloseTimedWith ::
+     MonadMonotonicTime m
+  => Tracer m EnclosingTimed
+  -> m a
+  -> m a
```

Comment:
There might be variants of this combinator, also see https://github.com/IntersectMBO/ouroboros-network/pull/3752#discussion_r876038163. I don't think we should do anything about this here; we can add more combinators here on a per-need basis.